### PR TITLE
Disable spawning of non-critical roles

### LINKED ISSUE #818: Stop Spawning Non-Critical Roles when Bucket <= CRITICAL_BUCKET_LEVEL * 2/3
Currently well below `CRITICAL_BUCKET_LEVEL`, working on up, but my spawns are working on non-critical roles instead of critical ones that are needed to stay afloat.

While `Game.cpu.bucket <= CRITICAL_BUCKET_LEVEL * 2/3` spawns should skip over non-critical roles.

### PR DESCRIPTION
**Pull request by Spedwards**
Resolves #818

### CODE REVIEW

**Comment by chobobobo** [comment]
Currently there is a bug with tasks that they don't track their queued creeps to see if they actually spawn or not, unfortunately this would hit on that bug big time.

Also, anyone that queues a creep will need to check to see if that creep is in the CRITICAL_ROLES list before queueing it to prevent thrashing adding/removing.

**Comment by Spedwards** [comment]
@chobobobo Not a perfect change just yet, but latest commit will keep the creeps in queue, but still skip over them.

**Comment by chobobobo** [comment]
Both @Spedwards and I have contributed to this one, so we're not in a position to review

File: spawn.js [hunk 1]
```diff
@@ -39,6 +39,7 @@ mod.extend = function(){
     };
     Spawn.prototype.createCreepByQueue = function(queue, level){
         if (!queue) return null;
+        else if (_.isUndefined(this.memory.spawnDelay)) this.memory.spawnDelay = {};
```
Comment (Spedwards) [inline spawn.js]: This whole line could be `Util.set(this.memory, 'spawnDelay', {});` and then the next line down can be changed to an `if`.

**Comment by karlthepagan** [comment]
Interesting to consider a few cases where critical roles are different from critical spawning:

* upgrader should only be a critical spawn if room.storage.sum > 90% capacity
* mineralMiner should only be a critical spawn if the mineral's renewal is < 10000 ticks (and storage.sum < 90% capacity)
* claimer should never be critical to spawn
* renewer should only be a critical spawn if their assigned room is < 300 reservation

**Comment by karlthepagan** [comment]
ran this for 12h and nothing died on me, bucket is 10k

will try to push my bucket by adding another SK room and file new PR's on any critical issues

**Pull request merged**